spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskNode.java

```diff
@@ -0,0 +1,336 @@
+		}
+		return s.toString();
+	}
+
+	/**
+	 * @return all the apps referenced in the composed task.
+	 */
+	public List<TaskApp> getTaskApps() {
+		if (taskApps == null) {
+			TaskAppsCollector collector = new TaskAppsCollector();
+			accept(collector);
+			taskApps = collector.getTaskApps();
+		}
+		return taskApps;
+	}
+	
+	/**
+	 * Walk the AST for the parsed composed task, calling the visitor for each element of interest.
+	 * See the {@link TaskVisitor} for all the events that can be received, since that is an
+	 * abstract class it can be extended and subclasses only need override methods to receive the
+	 * events of interest.
+	 * @param visitor a visitor to be called as the AST is walked
+	 */
+	public void accept(TaskVisitor visitor) {
+		Assert.notNull(visitor,"visitor expected to be non-null");
+		visitor.startVisit(this.name, this.composedTaskText);
+		int sequenceNumber = 0;
+		for (LabelledTaskNode ctn: sequences) {
+			if (visitor.preVisitSequence(ctn, sequenceNumber)) {
+				ctn.accept(visitor);
+				visitor.postVisitSequence(ctn, sequenceNumber);
+			}
+			sequenceNumber++;
+		}
+		visitor.endVisit();
+	}
+	
+	/**
+	 * @return this AST converted to a Graph form for display by Flo
+	 */
+	public Graph toGraph() {
+		GraphGeneratorVisitor ggv = new GraphGeneratorVisitor();
+		this.accept(ggv);
+		return ggv.getGraph();
+	}
+	
+	/**
+	 * @return a list of problems found during validation, empty if no problems
+	 */
+	public List<TaskValidationProblem> validate() {
+		TaskValidatorVisitor validator = new TaskValidatorVisitor();
+		this.accept(validator);
+		return validator.getProblems();
+	}
+	
+	/**
+	 * Simple visitor that discovers all the tasks in use in the composed task definition.
+	 */
+	class TaskAppsCollector extends TaskVisitor {
+
+		private String composedTaskName;
+		
+		private List<TaskApp> taskApps = new ArrayList<>();
+		
+		@Override
+		public void startVisit(String name, String composedTaskText) {
+			composedTaskName = name;
+		}
+		
+		@Override
+		public void visit(TaskAppNode taskApp) {
+			taskApps.add(new TaskApp(composedTaskName, taskApp));
+		}
+		
+		@Override
+		public void visit(TransitionNode transition) {
+			if (transition.isTargetApp()) {
+				taskApps.add(new TaskApp(composedTaskName, transition.getTargetApp()));
+			}
+		}
+		
+		public List<TaskApp> getTaskApps() {
+			return taskApps;
+		}
+
+	}
+	
+	public String getName() {
+		return this.name;
+	}
+	
+	public String getTaskText() {
+		return this.composedTaskText;
+	}
+	
+	/**
+	 * Shortcut to return the first node in the first sequence. Many composed tasks
+	 * will have just one sequence so do not force the consumer to dig through that
+	 * sequence.
+	 * 
+	 * @return the first node in the first sequence.
+	 */
+	public LabelledTaskNode getStart() {
+		if (sequences.size()==0) {
+			return null;
+		}
+		else {
+			return sequences.get(0);
+		}
+	}
+	
+	public List<LabelledTaskNode> getSequences() {
+		return sequences;
```

Comment:
mm, ok. the whole notion of sequences may disappear depending on where we go with DSL design post next release.